### PR TITLE
Feature/GCI-1180: Service refactor

### PR DESCRIPTION
**Pull request by mwejuli**
Project restructure to incorporate certified-copies user journeys.

### CODE REVIEW

File: src/controllers/certificates/delivery.details.controller.ts [hunk 1]
```diff
@@ -24,12 +24,14 @@ const logger = createLogger(APPLICATION_NAME);
 
 export const render = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
     try {
+      console.log('CERTIFICATE_DELIVERY_DETAILS');
```
Comment (nabeelamirch) [inline src/controllers/certificates/delivery.details.controller.ts:27]: remove these console.logs

File: src/controllers/certificates/delivery.details.controller.ts [hunk 2]
```diff
@@ -243,7 +247,7 @@ const route = async (req: Request, res: Response, next: NextFunction) => {
         logger.info(`Patched certificate item with delivery details, id=${req.params.certificateId}, user_id=${userId}, company_number=${certificatePatchResponse.companyNumber}`);
         await patchBasket(accessToken, basketDeliveryDetails);
         logger.info(`Patched basket with delivery details, certificate_id=${req.params.certificateId}, user_id=${userId}, company_number=${certificatePatchResponse.companyNumber}`);
-        return res.redirect(CHECK_DETAILS);
+        return res.redirect('check-details');
```
Comment (nabeelamirch) [inline src/controllers/certificates/delivery.details.controller.ts:250]: run lint to fix errors

File: src/controllers/certified-copies/check.details.controller.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import { addItemToBasket, getCertificateItem, getBasket } from "../../client/api.client";
+import { CHS_URL, APPLICATION_NAME } from "../../config/config";
+import { CertificateItem, ItemOptions } from "ch-sdk-node/dist/services/order/item/certificate/types";
+import { CERTIFIED_COPY_CHECK_DETAILS } from "../../model/template.paths";
+
+import { CERTIFIFIED_COPY_DELIVERY_DETAILS, replaceCertificateId } from "../../model/page.urls";
+import { Basket, DeliveryDetails } from "ch-sdk-node/dist/services/order/basket/types";
+import { createLogger } from "ch-structured-logging";
+
+const GOOD_STANDING = "Statement of good standing";
```
Comment (nabeelamirch) [inline src/controllers/certified-copies/check.details.controller.ts:12]: remove all unused variables and run lint

File: src/controllers/certified-copies/delivery.details.controller.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { Basket, BasketPatchRequest } from "ch-sdk-node/dist/services/order/basket/types";
+import { createGovUkErrorData, GovUkErrorData } from "../../model/govuk.error.data";
+import * as errorMessages from "../../model/error.messages";
+import { validateCharSet } from "../../utils/char-set";
+import { getAccessToken, getUserId } from "../../session/helper";
+// import { getCertificateItem, patchCertificateItem, getBasket, patchBasket } from "../../client/api.client";
+import { CERTIFIED_COPY_DELIVERY_DETAILS, CERTIFIED_COPY_CHECK_DETAILS } from "../../model/template.paths";
+import { createLogger } from "ch-structured-logging";
+import { APPLICATION_NAME } from "../../config/config";
+
```
Comment (nabeelamirch) [inline src/controllers/certified-copies/delivery.details.controller.ts:13]: run lint and remove all unused imports

File: src/controllers/certified-copies/home.controller.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { Request, Response } from "express";
+// import { CHS_URL, APPLICATION_NAME } from "../config/config";
+import { CERTIFIFIED_COPY_FILING_HISTORY, replaceCompanyNumber } from "../../model/page.urls";
+
+export default (req: Request, res: Response) => {
+    const companyNumber: string = req.params.companyNumber;
```
Comment (nabeelamirch) [inline src/controllers/certified-copies/home.controller.ts:6]: render empty page for this

File: src/middleware/certified-copies/auth.middleware.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { NextFunction, Request, RequestHandler, Response } from "express";
+import { SessionKey } from "ch-node-session-handler/lib/session/keys/SessionKey";
+import { SignInInfoKeys } from "ch-node-session-handler/lib/session/keys/SignInInfoKeys";
+
+// import { getCertificateItem } from "../../client/api.client";
```
Comment (nabeelamirch) [inline src/middleware/certified-copies/auth.middleware.ts:5]: removed unused imports

File: src/model/template.paths.ts [hunk 7]
```diff
@@ -1,7 +1,17 @@
-export const INDEX: string = "index";
+const certRoot = 'certificates';
```
Comment (nabeelamirch) [inline src/model/template.paths.ts:1]: double quote not single

File: src/model/template.paths.ts [hunk 8]
```diff
@@ -1,7 +1,17 @@
...
+const certifiedRoot = 'certified-copies';
```
Comment (nabeelamirch) [inline src/model/template.paths.ts:2]: call this certifiedCopyRoot

File: src/routers/certified-copies/routers.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { Router, Response, NextFunction, Request } from "express";
```
Comment (nabeelamirch) [inline src/routers/certified-copies/routers.ts:1]: remove unused express imports

File: src/controllers/certified-copies/check.details.controller.ts (hunk 3 above)
Comment (mwejuli) [reply src/controllers/certified-copies/check.details.controller.ts:12]: We can remove the unused imports as we work through the certified-copy stories seeing as we are tackling them next.. We can aim for perfection at that point. Or would you rather they were removed now and re-introduced as we work through the stories?

**Pull request merged**